llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -256,7 +256,7 @@ bool AArch64TTIImpl::areInlineCompatible(const Function *Caller,
     CalleeAttrs.set(SMEAttrs::SM_Enabled, true);
   }
 
-  if (CalleeAttrs.isNewZA())
```

Comment:
Should there be a similar restriction in clang (to ensure `always_inline`/`flatten` behave correctly)?

https://github.com/llvm/llvm-project/blob/42dfaa15a60cea6cd75a2efb419aa2c206d2a127/clang/lib/CodeGen/Targets/AArch64.cpp#L1201-L1203

Reply:
Thanks @MacDue, there should be a similar restriction in Clang which I've now added.